azure-pipelines.yml

```diff
@@ -109,6 +109,11 @@ stages:
             pytest: false
 
           - linux: py38-online
+            cache_dirs:
+            - key: 'sample_data'
+              path: .tox/py38-online/sample_data
+            - key: 'working_dir'
+              path: .tox/py38-online/tmp/sunpy
```

Comment:
I am not sure we want to cache this for the test builds, it might mean the tests skip downloads when they should test them etc.

Reply:
This is of course the point of #4831, I would expect that at least a few of our tests actually want to test transferring data, and not just parfive's skip-if-exist code :wink:

Reply:
Yeah, that makes sense. If there are specific tests that need to test downloading (and not just *use* the data) maybe there is a way to specify a different download directory for those tests, tell it to overwrite exiting, or simply delete certain files from the cached directory at the beginning of the test.

Reply:
Yeah, either setting `overwrite=True` or `path=tmpdir` to `Fido.fetch` would achieve those goals, but we would need to audit all the tests using Fido.fetch or other things making use of the default download dir to make sure we aren't footgunning ourselves.

Reply:
Shall we just put a cache on the doc build for now pending resolution of this?

Reply:
We can still cache the sample data, probably just not the working dir for now.